UnityProject/Assets/Scripts/Systems/Lobby/CharacterManager.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace Systems.Character
+{
+	/// <summary>Manage a player's characters. Intended for the local player client.</summary>
+	public class CharacterManager
+	{
+		public List<CharacterSheet> Characters { get; } = new();
+
+		private string OfflineStoragePath => $"{Application.persistentDataPath}characters.json";
+
+		public CharacterManager Init()
+		{
+			LoadOfflineCharacters();
+
+			return this;
+		}
+
+		public int GetCurrentCharacterIndex()
+		{
+			int lastCharacterIndex = PlayerPrefs.GetInt(PlayerPrefKeys.LastCharacterIndex);
+
+			return Math.Clamp(lastCharacterIndex, 0, Characters.Count);
+		}
+
+		/// <summary>Load characters that have been saved to the cloud.</summary>
+		public void LoadCharacters()
+		{
+			throw new NotImplementedException();
+		}
+
+		/// <summary>Load characters that are saved to Unity's persistent data folder.</summary>
+		public void LoadOfflineCharacters()
+		{
+			Characters.Clear();
+
+			if (!File.Exists(OfflineStoragePath)) return;
+
+			string json = File.ReadAllText(OfflineStoragePath);
+
+			var characters = JsonConvert.DeserializeObject<List<CharacterSheet>>(json);
+
+			if (characters == null) return;
+
+			foreach (var character in characters)
+			{
+				if (ValidateCharacterSheet(character))
```

Comment:
nesting can be reduced 
```cs
if (ValidateCharacterSheet(character) == false) continue;
Characters.Add(character);
```

Reply:
Thanks but I'd rather leave it as it is.